### PR TITLE
TA-04 Filter Tasks items

### PR DESCRIPTION
**Pull request by devankur03**
Filtering tasks data

### CODE REVIEW

File: src/components/taskArea/task-details/taskDetails.tsx [hunk 1]
```diff
@@ -29,6 +34,37 @@ const getBorderColorByStatus = (status: string) => {
 const TaskDetails: FC<PropsDetauls> = ({
     taskData,
 }: PropsDetauls): ReactElement => {
+    const [taskSwitchStatus, setTaskSwitchStatus] = useState(false);
+    const dispatch = useDispatch();
+
+    const updateTaskStatusInfo = (status: string, isChanged: boolean) => {
+        updateTask(taskData.id, { status })
+            .then((res: any) => {
```
Comment (3ankur) [inline src/components/taskArea/task-details/taskDetails.tsx:42]: Please remove the consoles,  and i didn't see loader logic anywhere.

File: src/features/task-list/taskListSlice.ts [hunk 2]
```diff
@@ -36,8 +40,76 @@ export const taskListSlice = createSlice({
...
+                if (action.payload?.status?.toLowerCase() === 'todo') {
+                    state.taskStatus.todo = (
+                        parseInt(state.taskStatus?.todo as string) + 1
+                    ).toString();
+                } else if (
+                    action.payload?.status?.toLowerCase() === 'inprogress'
+                ) {
+                    state.taskStatus.inProgress = (
+                        parseInt(state.taskStatus?.inProgress as string) + 1
+                    ).toString();
```
Comment (3ankur) [inline src/features/task-list/taskListSlice.ts:76]: Kind of repetitive logic we are using for both functions , can we move this logic to utility function

**Review by 3ankur — COMMENTED** [review]
Please fix those review comments.

File: src/features/task-list/taskListSlice.ts (hunk 2 above)
Comment (devankur03) [reply src/features/task-list/taskListSlice.ts:75]: Yea almost, we 'll refactor this in the new feature.

**Review by 3ankur — APPROVED** [review]
Looks good to me

**Pull request merged**